### PR TITLE
Add README and basic documentation

### PR DESCRIPTION
**Pull request by dbmtv**

### CODE REVIEW

File: README.md
Comment (Matovidlo) [inline README.md]: ?? Put there real reference not nats one

File: README.md
Comment (Matovidlo) [inline README.md:12]: Should be capable of everything in my opinion. Making new PR makes the documentation either old or have to be updated with it. However we dont support sometimes all of the calls on each entity. 
Please refer to code and docs that we cover it fully. If not this section is useless.

File: README.md
Comment (Matovidlo) [inline README.md]: Useless section, license file is included.

File: interaction_rating.go
Comment (Matovidlo) [inline interaction_rating.go:26]: I suppose the timestamp for delete serves for the different delete API calls the same. But in other section this is not included. 

I think here is point where the user should refer to `request_options.go` or official API reference, where this info is given.

File: item_values.go
Comment (Matovidlo) [inline item_values.go:8]: (some) is misleading. Either remove or change description

File: items.go
Comment (Matovidlo) [inline items.go:21]: ?? How it is possible that they are deted in cascade? I think it is only true, when `WithCascade` request option is given. I would not believe that it works as it is written here

File: recommendations.go
Comment (Matovidlo) [inline recommendations.go:18]: This is rather our use case

File: request_option.go
Comment (Matovidlo) [inline request_option.go:18]: Why there are not commentaries? This would be nice if it is provided. At least for some of them, but I suggest you to add text above each of the functions

File: series.go
Comment (Matovidlo) [inline series.go:8]: Here is missing info about seasons. They could be interpreted as series -> series -> items. It is important part how to distinguish between seasons and create better recommendation for example for new season (series of series) appearing in series container.

File: series.go
Comment (Matovidlo) [inline series.go:18]: If you use `WithCascade` it will be removed, I think. Check it in documentation and add it here or in `WithCascade` options description

File: series_items.go
Comment (Matovidlo) [inline series_items.go:8]: missing season description.

File: series_items.go
Comment (Matovidlo) [inline series_items.go:25]: again season description missing

File: series_items.go
Comment (Matovidlo) [inline series_items.go:39]: also it is possible to list series of series (season) elements. In such case series list returns series of series (seasons)

File: users.go
Comment (Matovidlo) [inline users.go:19]: You sure?

**Review by Matovidlo — CHANGES_REQUESTED** [review]
@dbmtv I added commentaries for particular parts that I am sure that are incorrect. However I do not approve this PR as the previous PR was closed without any reasonable comment. 

I would like to finish my PR at least in state that you started from, and merge this PR subsequently.

Thanks for understanding

File: README.md
Comment (dbmtv) [reply README.md]: My bad. Copy & paste.

File: README.md
Comment (dbmtv) [reply README.md:12]: This is the state we are at, it's good practice to mention that not all API endpoints are covered. Furthemore recombee may decide to add new endpoints and before we notice it, the client will be out of sync. It's a common practice to pinpoint exactly what the features are.

File: README.md
Comment (dbmtv) [reply README.md]: Check some other projects. It's common to refer to a LICENSE file.

File: item_values.go
Comment (dbmtv) [reply item_values.go:8]: It's copied verbatim from the official documentation: https://docs.recombee.com/api.html#set-item-values

File: items.go
Comment (dbmtv) [reply items.go:21]: Copied verbatim from the official documentation: https://docs.recombee.com/api.html#delete-item

File: recommendations.go
Comment (dbmtv) [reply recommendations.go:18]: Copied verbatim from the official documentation: https://docs.recombee.com/api.html#recommend-items-to-user

I'm fine with removing this paragraph as it may be too verbose for out purposes.

File: request_option.go
Comment (dbmtv) [reply request_option.go:18]: The problem is that these may be used in different contexts (for different API calls) and may have different connotation. I have some ideas how to improve it but better discuss it in person.

File: series.go
Comment (dbmtv) [reply series.go:8]: Copied verbatim from the official documentation: https://docs.recombee.com/api.html#add-series

**Comment by dbmtv** [comment]
> However I do not approve this PR as the previous PR was closed without any reasonable comment.

It wasn't possible to assign you as a reviewer to a PR which you yourself have opened.

File: README.md
Comment (dbmtv) [reply README.md]: I've removed it.

File: README.md
Comment (Matovidlo) [reply README.md:12]: I am able to approve it only if you check that we cover all of the mentioned sections fully, check it one more time.
1. Items, we do not implement `Delete More Items`
2. Item properies, we do not implement `List Item Properties`
3. Item properties, we do not implement the `Update More Items` call etc.

Please be so kind, go through official docs and do not say that it is feature that we implemented as it is not fully covered. 

I rather do not include it or include general case, that we do not fully cover the features. 

I would focus on implementing the calls in next PR and update docs of features once it is fully covered.

File: item_values.go
Comment (Matovidlo) [reply item_values.go:8]: I dont approve event official documentation, remove it.

File: items.go
Comment (Matovidlo) [reply items.go:21]: Why do we have copied docs?

I do not really understand the reason behind it. Rather we can have no docs and just push reference to this docs.

File: recommendations.go
Comment (Matovidlo) [reply recommendations.go:18]: Try to change it. This is official documentation, refer it as reference, instead of copying it

File: series.go
Comment (Matovidlo) [reply series.go:8]: I would rather push there the reference than copy it.

However, the official documentation is generic. Common use case for series->season->episode tree is to use this call 2 times and I rather mention it as we did the way through than rely on official documentation.

**Comment by Matovidlo** [comment]
I do not understand it. So you were not able to create new PR that is referenced to this one, so we fix only wrong parts? Then have 2 commits, 1 mine, 1 yours and merge it to master after both are finished?

Of course you can do it to this branch, but you had to close it and create new. 

In general I do not approve this kind of workflow and I would like to have discussion about it. We do not do such mess in our git, so why here in public?

**Review by Matovidlo — COMMENTED** [review]
Change the Features in Readme, otherwise it is fine.

File: README.md
Comment (dbmtv) [reply README.md:12]: I put a full list of supported features in the README.

Comment (Matovidlo) [reply README.md:12]: Nicely done

**Review by Matovidlo — APPROVED** [review]

**Pull request merged**
